Dev-Corner/slides/tool_integration.html

```diff
@@ -45,65 +45,1274 @@
+
+                            ```xml
+                            <command><![CDATA[
+                              #for $i, $s in enumerate( $series )
+                                rank_of_series=$i
+                                input_path=${s.input}
+                                x_colom=${s.xcol}
+                                y_colom=${s.ycol}
+                              #end for
+                            ]]></command>
+
+                            <inputs>
+                              <repeat name="series" title="Series">
+                                <param name="input" type="data" format="tabular" label="Dataset"/>
+                                <param name="xcol" type="data_column" data_ref="input" label="Column for x axis"/>
+                                <param name="ycol" type="data_column" data_ref="input" label="Column for y axis"/>
+                              </repeat>
+                            </inputs>
+                            ```
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### `outputs`
+                            #### What files the tool will produce as output?
+
+                            <img src="../images/output.png" alt="outputs" height=550/>
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### `outputs`
+
+                            ```xml
+                            <outputs>
+                                <data format="txt" name="tree" label="${tool.name} on ${on_string}:
+                                Tree" />
+                                <data format="txt" name="annotation" label="${tool.name} on
+                                ${on_string}: Annotation" />
+                            </outputs>
+                            ```
+
+                        </script>
+                    </section>
+
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### `outputs` > `filter`
+                            #### Conditional
+
+                            ```xml
+                            <outputs>
+                                <data format="png" name="png_output_image" label="${tool.name} on
+                                ${on_string}: Image">
+                                    <filter>format=="png"</filter>
+                                </data>
+                            </outputs>
+                            ```
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### `stdio`
```

Comment:
For stdio, there is the newish detect_errors param also: https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#command-tag